pubpub/pubpub
# enhancement: inline math editor

Implements #1917

See #1869 for further reference.
Uses [prosemirror-math](https://github.com/benrbray/prosemirror-math)


## Features

+ clicking the old equation formatting bar button will insert an empty `math_inline` node at the cursor's position
+ Typing `$stuff$` will result in an `math_inline` node.
+ Typing `$$` then space will result in a new `math_display` node.
+ Old `equation` nodes are converted at document load to new `math` nodes.

## Known places for improvement
+ after clicking the equation formatting bar button to insert an empty `math_inline` node, the cursor cursor should be placed inside the newly created node
+ after beginning a line of prose, typing `$$ ` causes the preceding text, which is in the same block as the newly-written double $, to be converted to a `math_display` node. Preferably, it would insert a `math_display` node _after_ the prose.

## Testing

Try typing with it, see how entering and exiting the math nodes feels. For copying in old `equation` nodes to test the node translation plugin, you can draw from [this reference pub on duqduq](https://demo.duqduq.org/pub/pxbv0zyj/draft?collectionSlug=math)

Try out the following exercises w/ new editing

+ create an empty testing pub
+ write an inline math node
+ write a display math node
+ copy/paste equation nodes from an old pub (try a mix of inline node, display node, and a selection of text that includes both as well as other node types)
+ try clicking into/out of math nodes
+ try using arrow keys to navigate into and out of math nodes
+ export math-laden pub to PDF
